esphome/components/sen5x/sensor.py

```diff
@@ -151,7 +151,7 @@
             cv.Optional(CONF_TEMPERATURE_COMPENSATION): cv.Schema(
                 {
                     cv.Optional(CONF_OFFSET, default=0): cv.float_,
-                    cv.Optional(CONF_NORMALIZED_OFFSET_SLOPE, default=0): cv.percentage,
+                    cv.Optional(CONF_NORMALIZED_OFFSET_SLOPE, default=0): cv.float_,
```

Comment:
Why not just allow negative percent?
```suggestion
                    cv.Optional(CONF_NORMALIZED_OFFSET_SLOPE, default=0): cv.possibly_negative_percentage,
```

Reply:
That would work, I'm not sure why it was percentage in the original to be honest, personally I find percentage a bit more confusing since the datasheet does not use percentage to refer to the slope so you have to mentally convert the factor of 100. I didn't know about `possibly_negative_percentage`, so I have no real objections to using it if it would be better for other reasons, it would mean the documentation PR (which just removes the words "in percentage") would not be required.

Reply:
The main reason for not changing it from percentage to float would be breaking existing configurations.
If people currently have eg `20%` in config, they will get an invalid after update and have to change it to `0.2` manually.

Reply:
One further potential benefit of `float` over `possibly_negative_percentage` is that it would be possible to support values larger than (+1/-1), which would be valid slope values but are not supported by `possibly_negative_percentage` (just discovered this when implementing the change).

The point about not breaking configs is valid and important, however, given that the temperature compensation didn't really work without this PR I'm not sure if anyone will have these values in their config (or at least it won't really be working if it is). On the other hand a slope larger than +1/-1 might also be unlikely. I'm happy to go either way but would lean towards removing the extra complication of percentage.

Reply:
I see, if its valid for it to go above 100% then changing to `float_` is acceptable.
Can you write a custom validation though that checks for people with a `%` and suggest the value to change it to with a `cv.Invalid`?

Reply:
I've added a custom validation.

As I say, I doubt there is anyone with a config including a percentage that they are currently happy with, since that config value was not working at all without this PR, but good to be thorough, and I've learnt a little more about how esphome works.